scalatest.dotty/src/main/scala/org/scalatest/CompileMacro.scala

```diff
@@ -162,8 +162,10 @@ object CompileMacro {
   }
 
   // check that a code snippet does not compile
-  def assertNotCompileImpl(compileWord: Expr[CompileWord], pos: Expr[source.Position])(shouldOrMust: String)(implicit refl: Reflection): Expr[Assertion] = {
+  def assertNotCompileImpl(self: Expr[Matchers#AnyShouldWrapper[_]], compileWord: Expr[CompileWord], pos: Expr[source.Position])(shouldOrMust: String)(implicit refl: Reflection): Expr[Assertion] = {
```

Comment:
@cheeseng When I was implementing compile macros, I found that the parameter `compileWord` is not used in macro implementation. It seems it can be removed both in Dotty & Scala2 macros.